src/plugins/intel_gpu/src/graph/reorder.cpp

```diff
@@ -276,6 +276,11 @@ void reorder_inst::update_output_memory() {
     if (_node != nullptr)
         build_deps();
 
+    // Do not update output memory when reorder is optimized out
+    // but input memory is not allocated yet because input is dynamic.
+    if (get_input_layout().is_dynamic())
```

Comment:
Shouldnt we check whether dep memory is allocated or not, instead of the layout? i.e., input_memory_ptr() == nullptr

Reply:
Done.